### DISCUSSION TITLE
Next big feature poll #4

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#659] [Polls]
Hello, just wanted to run a poll to gather opinions of what next big feature would be most helpful.

**TOP 2 features will have best chances to get implemented**

**Comment by MckenzieDavis** [comment]
I don't see why people would want GPU Undervolting if you can do it in MSI Afterburner

**Reply by seerge** [reply]
I also don't understand :) but lots of people seem to be asking. So I have agreed to at least look into that

**Reply by MckenzieDavis** [reply]
Im really² hoping that i wouldn't conflict with my config using Afterburner.

**Reply by seerge** [reply]
Do clock offsets conflict with afterburner? cause they should set same setting and be read by afterburner even

**Reply by Smokyyy** [reply]
Oh please. I'd rather not have MSI Afterburner installed too. I already got ptsd from all this cringe and ugly "gamery" bloatware from Asus, no need MSI too. I like your sleek design without useless "gamery" look which is widespread on gaming laptops. Anyway, please include Nvidia GPU undervolting if AMD is already done, no need to exclude us :).

**Reply by R4333** [reply]
MSI Afterburner is an excellent piece of software and has a very usable UI. I don't understand your concerns.

**Reply by Provinomico** [reply]
I've been an afterburner user for over a decade, but I'm a fan of not having multiple tools running for specific things, if you can control clocks and voltages in G-Helper I'd be more than happy to uninstall afterburner (or at least not having it running all the time, the monitoring graphs are still useful sometimes). BTW I thought undervolting on mobile RTXs was locked anyway.

**Reply by MckenzieDavis** [reply]
Im not quite sure if its conflict or not because i didn't overclock Core or Memory in either GHelper or Afterburner, i just use it to undervolt, will test when the undervolt feature is ready in GHelper.

**Reply by MckenzieDavis** [reply]
You can always set to apply the tune without having to open MSI all the time, just set it to apply at startup and untick running MSI Afterburner at startup, and the tune will apply even without running MSI Afterburner

**Reply by RahulVivekNair** [reply]
Also afterburner keeps the gpu active at all times and does not work well with nvidia optimus.

**Reply by MckenzieDavis** [reply]
> Also afterburner keeps the gpu active at all times and does not work well with nvidia optimus. 

I just replied the answer above you, you don't need to run MSI afterburner 24/7, it should still applied after you close it

**Reply by mmp5453** [reply]
the whole point of ghelper is to minimalize the amount of unecessary software. how on earth can you not comprehend why people would not want to use afterburner?

**Reply by Tezaphon** [reply]
Would be optimal to have undervolting for both GPUs in one program.  Considering that afterburner has some similar functionality with ghelper (fans speeds, temps, etc), would like to avoid having two programs with the same purpose.

**Reply by seerge** [reply]
@Tezaphon afterburner can't control fan speeds on laptops :)

**Reply by morgenstern09** [reply]
Indeed, it can't control fans. As I understand, G-Helper doesn't control fan either, but changes the fan curve. Based on that the fan will go faster/slower based on the temperature vs. RPM you set.

**Reply by joex92** [reply]
excuse my ignorance... what's undervolting for?

**Comment by Kickskii** [comment]
I have a suggestion to make,
I would like to suggest that undervolting options be only enabled after being "requested" to be enabled by the user through a check box in the extra labelled along the lines of "experimental feature use at your own risk" just seems better for there be at least one extra step involved in a process that can very easily end up in a boot loop if someone turned on the auto apply button with a high enough curve. It just doesnt seem right for it to just be there in plain sight like a red button with the label "do not touch".

Maybe you could also add a safety feature where it would disable applying of curve on boot after a crash since curve optimizer is extremely silicon lottery dependent and not everyone actually would have the knowledge to test out stability, maybe consider adding a stress test utility that tests out the cores? tbf that's asking too much im sorry
absolutely love the thermal limit placement tho

**Reply by seerge** [reply]
But this how it's done now.

You need to click "Apply" - which will apply UV, but not set it to be automatically ON
and only AFTER that "Auto Apply" checkbox unlocks

**Reply by Kickskii** [reply]
i can stay alive on -30 for a few minutes(if im not doing much which i assume will be the case for most when setting up ghelper) before windows crashes and i have absolutely terrible silicon
my core 6 fails ycruncher at 0

**Reply by Kickskii** [reply]
it being hidden behind one text box with a label will make sure people actually read the subtext 
i saw undervolt options and barely saw the subtext about it being risky since its a block of text and usually the brain would look at the shiny button first
maybe having the person read a small text first would be a good idea

also most people having access to undervolting consider the fact that they need to test the curve using stress test so maybe some way to educate them of that fact would be good?

**Reply by seerge** [reply]
Well, i don't know how to "detect" a crash in windows in a first place (and especially if that crash was cause of instability or just windows crashed by some unknown reasons). I would suggest not to click auto apply untill you are 100% sure that this is your stable settings with UV

**Reply by Kickskii** [reply]
i 100% agree with you on you suggesting to not clicking auto apply but you need to communicate this to the average person that doesnt know how curve optimizer works on amd and how having even minor instabality can eventually lead to data corruption and maybe a windows corruption where the only option is a reinstall

**Reply by seerge** [reply]
I can unlock "auto apply" checkbox not immediately but after some time (like 1 minute) and show countdown saying that you need to make sure system is stable :) but i'm not sure if this is not an overkill

**Reply by Kickskii** [reply]
i hope im making sense sorry for the long replies its just i very much like the tool but as someone who has explained to people how to tune their g15ae's most of them were completely new to the scene so people might just go for it and go "if it dont crash its fine" which is quite bad and they need to know the risks

edit: minor instabilities just end up with data corruption so somehow need to make sure that the people using the feature know what they getting into
edit 2: this usually happens within around a curve value delta of ~5 since i am sure i will not crash if the stability is within 5 points away sometimes 7
i can run -30 on my core 3 and not crash. doesnt mean i wont get errors for every single stresstest i throw at that core but i can do it

**Reply by seerge** [reply]
I have adjusted label text and made it much more prominent
![Screenshot 2023-06-22 144123](https://github.com/seerge/g-helper/assets/5920850/4e2e04fb-1ba1-4228-bc71-1aab8fd179a9)

**Reply by Kickskii** [reply]
that looks awesome! 
maybe throw in a link to some guide on the internet explaining how to test curve optimizer for example using core cycler with prime95 avx2 or ycruncher kagari presets with an underline(i think thats default for an hyperlink but i could be wrong just want to cover my base) and i think it should be perfect to get someone rolling with an all core curve optimizer setup
i also linked a excel sheet that has hex values for doing it on a per core level from ryzenadj on your last major feature poll in case youre using the same approach it might make it easier for you to implement it since most silicon if being held back is usually held back from one or two bad cores while other cores can take a lot higher curve

**Reply by ke1ne** [reply]
> I have adjusted label text and made it much more prominent !

I would gently propose to make it blink. =)

**Reply by Kickskii** [reply]
> I would gently propose to make it blink. =)

or have text in a larger font size in bold that says "warning" first since that sounds way easier

**Reply by seerge** [reply]
@Kickskii i think that's a bit too much, i can add a link to some guide in readme (but again it has to be something very simple and user-friendly), as some people literally struggle to unzip a file even.

As for per-core changes, it's also too much for the app, if someone want's to get very detailed tweaking they can use UXTU , which is better anyway :)  same as afterburner is better for overclocking / undervolting of GPU.

**Reply by seerge** [reply]
@ke1ne 
> I would gently propose to make it blink. =)

Yep, and also draw a big pink elefant on top :)

**Reply by Kickskii** [reply]
@seerge yeah that sounds appropriate a guide in the readme would be good and having a link under on some highlighted text to the same guide in program warning label just in case they missed it should be plenty for most people

i understand about per core since it can definitely get complicated very fast i just dont like uxtu personally something feels off about using it i cant explain it just a feeling that its incomplete and unpolished to the point i actually prefer using simple batch files as a more elegant solution to the curve optimizer problem as of writing this that is

**Reply by seerge** [reply]
@Kickskii do you have a good undervolting guide (understandable for a person struggling to unzip a file) in mind ?

**Reply by Kickskii** [reply]
@seerge i made one to do it on a per core basis when i made the tuning guide for the g15 advantage edition https://youtu.be/SUCBbKPeNDE?t=806 (unintentional plug i am really sorry) and you could maybe make one thats similar since i walk through it step by step in like five minutes but you will still have to write it down ive talked with quite a few people and they managed to do it using the guide so i would assume something like how i explained might work for the average joe

tho from experience something like this image im sharing is a prime example of what doesnt actually work since even some of the more savvy people felt this to be daunting to follow how i explained it in the below image so basically you might need to hold the end user's hand through it
the below also lets you test it out on a all core basis with the exception that you should still change it to either p95 avx or y-cruncher kagari
![Old Guide on how to do per core](https://github.com/seerge/g-helper/assets/56576955/cf762fc4-bcba-4b98-9ed6-31f304555772)
also PSA they should absolutely set the apu/cpu ppt/power limit to the maximum value they saw in armourycrate manual mode(tho i think ghelper is already reading that) AND having processor boost set to aggressive
time values for y-cruncher kagari is 10 minutes per core for four iterations for a total of 40 minutes per core
and prime95 avx2 huge ffts 60 minute per core
these are the minimum time limits i recommend one test for also i recommend they test both of them
y cruncher usually detects instability earlier but some have reported prime95 still finding errors after y cruncher tests
also i apologize for the information dump

**Reply by seerge** [reply]
@Kickskii that's great, but I meant a simple tutorial about what undervolting is, and why it's worth to try it :) Understandable for an average person who is absolutely not familiar with this things ;)

**Reply by Kickskii** [reply]
@seerge the thing is there is basically no "perfect" guide for the average joe that i know of that would explain it in a simple enough fashion I'm sure you are looking for, you will have to scrap things together from different sources and word them in a concise yet easy to understand and follow along way 
if you have a discord server or something we can work together to type one down for the average joe so they don't feel lost (if you even need my help for that, that is)
most guides I know of are usually paired with PBO limits (even i did this in my guide cause its part of the pbo tuning process) which add another variable to the explanations when you should have those values fixed before touching CO as any change to them **WILL** shift CO values and are basically unsuitable for explaining it to an average person with only a baseline understanding of these topics

**Reply by seerge** [reply]
I think LinusTechTips video could do a starter's job https://www.youtube.com/watch?v=9b3JUiK1aWU :)

**Reply by Kickskii** [reply]
>I think LinusTechTips video could do a starter's job https://www.youtube.com/watch?v=9b3JUiK1aWU :)

https://youtu.be/4o5AZM0AJXA?t=56 watch till 3:18 the ltt video isnt exactly correct for curve optimizer since its not actually undervolting unless you have a way of locking down clock speeds and many times on mobile the cores dont even reach the advertised boost speeds out of the box by default
i will give my own example my first four cores core 0,1,2,3 cannot go above 4.6-4.65ghz without curve optimizer at which point each of them manages to reach 4.75-4.79 while my remaining cores can boost to 4.9 to 4.95 this is reflected through my curve values on each cores as well since those "bad cores" can get a significantly higher curve value

**Reply by Kickskii** [reply]
also @seerge the core vid's are locked on mobile
you arent actually reducing those when you do curve optimizer you are adjusting the voltage frequency curve
if they were unlocked i would have been running 1.2v the moment i get the chance to do it
well tried is a better term but welp

**Comment by seerge** [comment]
CPU power limits for old devices with AMD (i.e. G14 2020) is live as https://github.com/seerge/g-helper/releases/tag/v0.90

**Reply by MckenzieDavis** [reply]
Can confirm it works on 4800hs!

**Reply by MckenzieDavis** [reply]
I think reapply for certain amount of time is required since the power limit tweak get resets by itself after a few moment, Like in UXTU you can set reapply at amount of time for example 2 seconds etc, i don't know what's causing this reset, but the way that UXTU implement this method suggest that there's no other way around to keep the power limit setting stays, i noticed this by playing game and after few minutes. It's kinda random, sometimes it's instant, sometimes it takes a few minutes.

**Reply by seerge** [reply]
Do you know how often it resets? as it doesn’t for me

**Reply by MckenzieDavis** [reply]
> Do you know how often it resets? as it doesn’t for me

Sometimes it takes a short time, sometimes a bit long, I'm monitoring using HWiNFO.

**Reply by ke1ne** [reply]
@seerge I'm agree with @MckenzieDavis, ryzenadj-based software I've used previously had an option to re-apply settings by timer. This can make sense

**Reply by seerge** [reply]
And then someone will accuse app of causing stutters or so :) cause low level settings are applied like every 2 seconds. I would like to know if this is really needed first and what resets params (if this happens) ? 

I have noticed that there is a so called "Ryzen master" app in task scheduler (that would run all the time). It's being added by amd driver installation. I suspect it could be also applying own stuff and therefore resetting some settings. Can you try to stop / disable it ?
![Screenshot 2023-06-24 113848](https://github.com/seerge/g-helper/assets/5920850/9dcc4713-9465-4f78-823c-2a5844542726)

**Reply by MckenzieDavis** [reply]
> And then someone will accuse app of causing stutters or so :) cause low level settings are applied like every 2 seconds. I would like to know if this is really needed first and what resets params (if this happens) ? 
> 
> I have noticed that there is a so called "Ryzen master" app in task scheduler (that would run all the time). It's being added by amd driver installation. I suspect it could be also applying own stuff and therefore resetting some settings. Can you try to stop / disable it ?
> ![Screenshot 2023-06-24 113848](https://github.com/seerge/g-helper/assets/5920850/9dcc4713-9465-4f78-823c-2a5844542726)
> 
> 

I believe for 4000 series there's no Ryzen Master's driver been installed in AMD chipset driver since im using latest version. Or i need to install ryzen master to get that?

**Reply by seerge** [reply]
You don't need to manually install that, do you have any other "suspicious" processes from amd / ryzen running that could set own settings?

**Reply by MckenzieDavis** [reply]
> You don't need to manually install that, do you have any other "suspicious" processes from amd / ryzen running that could set own settings? 

Nope, clean windows , only drivers related that i installed is amd display driver, amd chipset and nvidia drivers, it's a known issue as far as im concerned. I'll try find out what's the cause.

Stopped all amd process using Autoruns, so 0 amd processes are running

**Reply by ke1ne** [reply]
@seerge Ryzenmaster is compatible not with all CPUs and laptops. For instance, on G15 2021 (Ryzen 9 5900HS) it even wasn't able to install properly.

**Reply by seerge** [reply]
@ke1ne i'm not saying it's compatible :) but amd driver package just installed that thing for me :)

**Reply by ke1ne** [reply]
> @ke1ne i'm not saying it's compatible :) but amd driver package just installed that thing for me :)

Sure, didn't mean that too. So, Nvidia undervolting is still on the table, right?

**Reply by seerge** [reply]
It is :) but i don't have a device with nvidia atm, so can look into that when I will get one ...

**Reply by ke1ne** [reply]
@seerge I can run some your scripts to get API entry points, etc., if need

**Reply by seerge** [reply]
Does undervolting work for you in afterburner ? if so - do you have discord ?

**Reply by ke1ne** [reply]
> Does undervolting work for you in afterburner ? if so - do you have discord ?

Have no idea. =) Will install and test, a little bit later.

**Reply by ke1ne** [reply]
@seerge so, we can test. discrod is baker580

**Comment by seerge** [comment]
Added a button to stop all apps using GPU https://github.com/seerge/g-helper/releases/tag/v0.91

**Comment by MckenzieDavis** [comment]
> the whole point of ghelper is to minimalize the amount of unecessary software. how on earth can you not comprehend why people would not want to use afterburner?

I was pointing it out because I saw a lot of people don't know that you don't have to run afterburner all the time to keep it settings, while the added feature might contribute to adding resources usage in ghelper which will be running all the time, but if that the feature that majority want to have, who am I to argue.

**Comment by p-chandel** [comment]
Using text on the Anime Matrix is something that would be nice. It was the only thing I used the matrix for.

**Reply by seerge** [reply]
You can do that by putting text on a picture actually (even in mspaint) and then setting that picture:)

**Reply by joex92** [reply]
Is there a template for each laptop version? I have a G14 2021 edition... or is it the same for any?

**Comment by GeometricParrot** [comment]
Could you make G-Helper check for updates for itself when you open the 'updates' section?

**Reply by seerge** [reply]
App checks for own updates every time you open its window (but not more often than 12 hours).

I think it's more than enough:)

**Reply by GeometricParrot** [reply]
Ok

**Comment by Louis047** [comment]
> Hello, just wanted to run a poll to gather opinions of what next big feature would be most helpful.
> 
> **TOP 2 features will have best chances to get implemented**
Make GPU Undervolting easier like for beginners with a guide for each gpus and try to make it as simple as possible :)

**Comment by musm** [comment]
Top feature for me would be signed executable 😄  Not a fan of always having to click accept when g-helper starts from reboot.

**Reply by seerge** [reply]
Actually windows should remember your choice, and not ask every time. You can also right click -> open exe properties -> click Unblock there (which does same)

**Comment by seerge** [comment]
Small update on undervolting progress. It appeared that method to undervolt I had hopes for doesn't work, so for now I'm putting this on hold until I figure out other approaches :)

**Comment by Grail01** [comment]
First post. Just started using G-Helper after 6 months with flowOSD. Loving it so far.

One feature that would change G-Helper from awesome to "take my money", would be if it could wrestle control of sleep modes. Most Asus machines have got screwed up Suspend modes eg missing S3 and Windows 11 is pretty broken and put together, sleep modes just don't work. For me, at the moment, I can hibernate but not sleep, e.g. no fast restart. I figure this may well be out of the scope of what you are looking to achieve with this tool, but worth asking :)

**Reply by IceStormNG** [reply]
It actually does sleep quite reliably once you adjusted the windows power plan settings for this. Their defaults are crap, as they're designed for devices with low power CPUs, as which ASUS laptops usually do not classify.

This is a tool to adjust power plan settings even if they're hidden. It lists every single setting. Though not all do something depending on your platform.
https://forums.guru3d.com/threads/windows-power-plan-settings-explorer-utility.416058/

The ones to look out for are:

**Sleep -> Allow Wake timers** This allows tasks to wake up the PC in sleep to do background tasks. Definitely disable for battery, maybe leave on for AC if you like. Tasks like windows update, telemetry and other maintenance stuff runs here

**Sleep- > Hibernate After** This defines after how many seconds in sleep (S0) the machine will hibernate. Set to 0 to disable it.

**Sleep- > Allow Away Mode Policy** Allows or denies away mode. Away mode is a state where your screen is off but the machine is still running. Apps can request that and prevent going to proper sleep which drains the battery.

**Presence Aware Power Behavior  -> User Presence Prediction mode** tries to predict whether you're sitting in front of the PC or not through various sensors and metrics. Might prevent sleep or wakeup the machine for no good reason.

Also check  **powercfg /reqiuests** in an admin CMD for devices or services that request the device to be awake.
**powercfg -devicequery wake_armed** shows all devices that are configured to wake  up the PC. This often includes HID devices like mice or keyboards. You can disable that in the device manager under "Energy Preferences" of the respective device.

After adjusting those values, there are still potential 3rd party services or tasks that wake up the PC. You can find them though.
Open and admin CMD and run **powercfg /sleepstudy**. Open the generated HTML file and it gives you a report for every power mode change.
The first is the earliest one the last is the newest one. Green marked are sleep states with "good sleep" orange are "quite a lot wakeup" and red is "almost no sleep at all" Gray ones are non-sleep states and are usually not of interest here.

If you click on them you get a breakdown on what caused the sleep issues:
This here for examples shows that the PCIe Root port PEG2 (2nd NVMe slot) and USB4 Host controller somehow prevented sleep. So some app or USB device did some crap that prevented sleep. Can be a mouse that is connected and doesn't turn off properly and is spamming input to the PC. SSD means some process caused I/O on the SSD or the SSD failed to go to low power modes.

<img width="1038" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/a7cf2988-0af1-48b6-8d35-83746bbdf252">

Here you can also see two tasks. WU = Windows Update and Update Orchestration Client (also part of Windows update) were running and likely finished Windows updates.
<img width="1015" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/e83dce78-f928-4fbf-b1d1-0af6db3e7cdf">

Going further in, you see it was scanning for updates.
<img width="366" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/c0c5ad2d-68f4-42aa-8f80-79552d93e21f">

It takes a bit of time depending on how much stuff runs on your PC. HPs Printer software for example has a "Printer Doctor" service that wakes up the machine, even on battery for no fu*\*\* reason, even on battery. I had to disable that.

But after you did this dance once, you can get the machine to reliably sleep when closing the lid. Yes, this is actually something the OEM (ASUS) was supposed to pre-configure so it works at least out-of-the box, but lazy ASUS was too busy making even more crappy software that we now have to replace (lmao).

Also: I don't think GHelper needs any tools for that, at best a powershell script to change the power plan settings to more sane defaults, though it is best you adjust them yourself to your needs. Some people want some of the settings to be on when plugged in, others do not want to. Sleepstudy analysis is manual work anyways.

**Reply by Grail01** [reply]
Hi IceStormNG,

Amazing. Not sure anyone has ever taken the time to write me such a detailed and helpful response to a random question before. Thanks!

**Comment by Colyn2002** [comment]
I hope to reduce the power consumption of the graphics card, sometimes the notebook needs CPU performance more than the graphics card

**Comment by joex92** [comment]
is there any progress on the Anime Matrix editor?

**Reply by seerge** [reply]
@joex92 it is already live :) you can resize and position any image you want

**Comment by savi4587** [comment]
Is there any updates on adding an undervolt option to rtx GPUs ? (Emphasis on 40 series) ? You made such a great and all in one app that i don't wanna install anything else:)

**Reply by seerge** [reply]
@savi4587 you can already undervolt any nvidia GPU for ages :) 
https://github.com/seerge/g-helper/wiki/FAQ#how-to-undervolt-gpu

**Reply by savi4587** [reply]
oh, I didn't realize you could undervolt that way, tysm

**Comment by 0xKonstantin** [comment]
Honestly I would love to see the ability to edit the e and p cores from G-Helper. Needing this has me having to go back to armory crate as I don't wanna boot to bios every time I need to switch

**Reply by seerge** [reply]
@0xKonstantin that's already possible for ages for Intel models that support that in a first place (under `Extra` settings)

**Reply by 0xKonstantin** [reply]
Oh well shoot, I totally missed that and even searching here I didn't see mention so thought it didn't exist 

Thanks Seerge,
I will leave you a tip!
